libs/server/Objects/List/ListObjectImpl.cs

```diff
@@ -418,5 +419,211 @@ private void ListSet(ref ObjectInput input, ref SpanByteAndMemory output)
+                    {
+                        var nextNode = currentNode.Previous;
+                        if (currentNode.Value.AsSpan().SequenceEqual(element))
+                        {
+                            if (rank == -1)
+                            {
+                                lastFoundItemIndex = currentIndex;
+                                while (!RespWriteUtils.WriteInteger(currentIndex, ref output_currptr, output_end))
+                                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref output_startptr, ref ptrHandle, ref output_currptr, ref output_end);
+
+                                noOfFoundItem++;
+                                if (noOfFoundItem == count)
+                                {
+                                    break;
+                                }
+                            }
+                            else
+                            {
+                                rank++;
+                            }
+                        }
+                        currentNode = nextNode;
+                        currentIndex--;
+                    }
+                    while (currentNode != null && currentIndex >= maxlenIndex);
+                }
+
+                if (noOfFoundItem == 0)
+                {
+                    output_currptr = output_startptr;
+                    while (!RespWriteUtils.WriteNull(ref output_currptr, output_end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref output_startptr, ref ptrHandle, ref output_currptr, ref output_end);
+                }
+                else if (count > 1 && noOfFoundItem == 1)
+                {
+                    output_currptr = output_startptr;
+                    while (!RespWriteUtils.WriteInteger(lastFoundItemIndex, ref output_currptr, output_end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref output_startptr, ref ptrHandle, ref output_currptr, ref output_end);
+                }
+                else if (noOfFoundItem > 1)
+                {
+                    var newTotalArrayHeaderLen = 0;
+                    var startOutputStartptr = output_startptr;
+                    RespWriteUtils.WriteArrayLength(noOfFoundItem, ref startOutputStartptr, output_end, out var _, out newTotalArrayHeaderLen);  // ReallocateOutput is not needed here as there should be always be available space in the output buffer as we have already written the max array length
+                    Debug.Assert(totalArrayHeaderLen >= newTotalArrayHeaderLen, "newTotalArrayHeaderLen can't be bigger than totalArrayHeaderLen as we have already written max array lenght in the buffer");
+
+                    if (totalArrayHeaderLen != newTotalArrayHeaderLen)
+                    {
+                        Buffer.MemoryCopy(output_startptr + totalArrayHeaderLen, output_startptr + newTotalArrayHeaderLen, output.Length, output.Length);
```

Comment:
Only remaining length after array size data needs to be copied

Reply:
I am not sure I get this. Am I not already doing that using this code `output_startptr + totalArrayHeaderLen`?

Reply:
Shouldn't the length parameters be adjusted to exclude the bytes corresponding to the array length header as only the remaining bytes will have to be moved?

Reply:
Ohh got it my bad, I will fix it.

Reply:
Fixed it